bcrypt.js

```diff
@@ -152,15 +152,15 @@ module.exports.compareSync = function compareSync(data, hash) {
 /// @param {String} hash expected hash
 /// @param {Function} cb callback(err, matched) - matched is true if hashed data matches hash
 module.exports.compare = function compare(data, hash, cb) {
-    if (data == null || hash == null) {
+    if (typeof data === 'function') {
         return process.nextTick(function() {
-            cb(new Error('data and hash arguments required'));
+            data(new Error('data and hash arguments required'));
```

Comment:
There is no check now in this function for `(data == null || hash == null)` when no callback is provided (also in the hash function). Is this not going to be an issue?